master/buildbot/steps/master.py

```diff
@@ -242,3 +246,37 @@ def start(self):
         self.addCompleteLog(name='Output', text=content)
         self.step_status.setText(self.describe(done=True))
         self.finished(SUCCESS)
+
+
+class CancelRelatedBuilds(BuildStep):
+    name = 'CancelRelatedBuilds'
+    description = ['Checking']
+    descriptionDone = ['Checked']
+
+    def __init__(self, isRelevant=None, builderNames=None, **kwargs):
+        BuildStep.__init__(self, **kwargs)
+
+        if isRelevant is None:
+            config.error('You must provide a function to check '
+                         'if a build is relevant')
+        self._is_relevant = isRelevant
+
+        if builderNames is None:
+            self._builder_names = self.master.botmaster.builderNames[:]
+        else:
+            self._builder_names = builderNames
+
+    @defer.inlineCallbacks
+    def run(self):
+        control = interfaces.IControl(self.master)
+
+        for name in self._builder_names:
+            builder_control = control.getBuilder(name)
+
+            pending = yield builder_control.getPendingBuildRequestControls()
```

Comment:
In what way?

If the user wants to only cancel builds, she'd use this step.  If she also wants to stop builds, she'd add `StopRelatedBuilds` step.

I see that the recommended way will be to first use `CancelRelatedBuilds` and then `StopRelatedBuilds`...